app/routes/extension/DownloadApps/containers/DownloadApps.js

```diff
@@ -78,10 +79,11 @@ class DownloadApps extends Component {
       showQrIos,
       showQrPairing
     } = this.state
-    // const { safes } = this.props
+    const { safes } = this.props
 
-    // if (safes != null && safes.safes.length > 0)
-    //  return <Redirect to='/account' />
+    if (safes !== null && safes.safes.length > this.pairedSafes) {
```

Comment:
```safes.safes```  should be improved. Same as the first check. That should come from the props.

Reply:
Also:
```
 <Layout
        toggleQrAndroid={this.toggleQrAndroid}
         toggleQrIos={this.toggleQrIos}
         toggleQrPairing={this.toggleQrPairing}
         showQrAndroid={showQrAndroid}
         showQrIos={showQrIos}
         showQrPairing={showQrPairing}
         androidAppLink={this.androidAppLink}
         iosAppLink={this.iosAppLink}
         password={this.password}
       />
```
Despite of I understand what you are trying to do, if at the end you are using "constants" for the links, it is absolutely fine if you just import them in children components. In that way you could reduce the props-descendant-hell a little bit.